src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/AbstractBitbucketEndpoint.java

```diff
@@ -55,16 +66,42 @@
     @CheckForNull
     private final String credentialsId;
 
+    /**
+     * Jenkins Server Root URL to be used by that Bitbucket endpoint.
+     * The global setting from Jenkins.getActiveInstance().getRootUrl()
+     * will be used if this field is null or equals an empty string.
+     * This variable is bound to the UI, so an empty value is saved
+     * and returned by getter as such.
+     */
+    private final String bitbucketJenkinsRootUrl;
```

Comment:
Should not be final

Reply:
My understanding was that `final` is used for configuration values, so effectively set once per object lifetime (a reconfig creates a new object with same `serverUrl` for consumers to reference, and removes the current one), and the keyword enforces that?

Reply:
it should not be marked final since it can be changed after the fact when the config is created.